Packs/OracleCloudInfrastructure/Integrations/OracleCloudInfrastructureEventCollector/OracleCloudInfrastructureEventCollector.py

```diff
@@ -0,0 +1,387 @@
+            str: Base URL for the client.
+        """
+        if not is_region(region):
+            raise DemistoException('Could not create a valid OCI configuration dictionary fou to invalid region parameter. \
+                Please check your OCI related instance configuration parameters.')
+
+        return f'https://audit.{region}.oraclecloud.com/{PORT}/auditEvents'
+
+    def validate_private_key_syntax(self, private_key_parameter: str) -> str:
+        """Validate private key parameter syntax.
+        The Private Key parameter needs to be provided to the OCI SDK singer object in a specific format.
+        The most common way to obtain the private key is to download a .pem file from the OCI console.
+        If copied from a .pem file, the private key parameter may contain unnecessary spaces.
+        Further more, since the format uses \n as part of the key,
+        passing this value as a configuration parameter may result in escaped \n characters.
+
+        This function will preform the following actions on the private key parameter:
+            - Unescape the string.
+            - Remove unnecessary spaces in the string.
+
+        Example:
+        Raw Private Key parameter: -----BEGIN PRIVATE KEY-----\\n\n THIS-IS-A\\n\n PRIVATE-KEY\\n\n -----END PRIVATE KEY-----
+        Output: -----BEGIN PRIVATE KEY-----\nTHIS-IS-A\nPRIVATE-KEY\n-----END PRIVATE KEY-----
+
+        Args:
+            private_key (str): Private Key parameter.
+
+        Returns:
+            str: Private Key parameter unescaped and spaceless.
+        """
+        private_key = stringUnEscape(private_key_parameter)
+        private_key = private_key.replace('\n\n', '\n')
+
+        if ' ' not in private_key:
+            return private_key
+
+        prefix = '-----BEGIN PRIVATE KEY-----\n'
+        postfix = '\n-----END PRIVATE KEY-----'
+
+        private_key = private_key.replace(prefix, '').replace(postfix, '')
+
+        private_key_sections = private_key.strip().split(' ')
+        striped_private_key = ''.join(private_key_sections)
+        return prefix + striped_private_key + postfix
+
+
+''' Event related functions '''
+
+
+def add_time_key_to_events(events: List[dict[str, Any]]) -> List[dict[str, Any]]:
+    """
+    Add the _time key to the events.
+    Args:
+        events (List[dict[str, Any]]): The events to add the time key to.
+    Returns:
+        List[dict[str, Any]]: The events with the _time key.
+    """
+    for event in events:
+        if event_time := event.get("eventTime"):
+            event["_time"] = event_time
+
+    return events
+
+
+def get_last_event_time(events: List, first_fetch_time: datetime) -> str:
+    """Get the latest event time from the fetched events list for next fetch cycle.
+    - Given a non empty list of events, the function will return the time of the last event (most recent event) + 1 milliseconds.
+    - If the event list is empty, the function will return the current first fetch time.
```

Comment:
```suggestion
```
We can remove it from here cause you added it in the returns section.

Reply:
I prefer to leave the doc string as is and remove the matching comment if its the same to you.

Reply:
Sure, np :)